language/compiler/ir-to-bytecode/syntax/src/syntax.rs

```diff
@@ -62,31 +62,49 @@ fn consume_token<'input>(
     Ok(())
 }
 
-fn parse_comma_list<'input, F, R>(
+fn parse_comma_list<'input, F, G, R>(
     tokens: &mut Lexer<'input>,
-    list_end_token: Tok,
+    is_end_of_list: G,
     parse_list_item: F,
     allow_trailing_comma: bool,
 ) -> Result<Vec<R>, ParseError<usize, anyhow::Error>>
 where
     F: Fn(&mut Lexer<'input>) -> Result<R, ParseError<usize, anyhow::Error>>,
+    G: Fn(&Lexer<'input>) -> bool,
 {
     let mut v = vec![];
-    if tokens.peek() != list_end_token {
+    if !is_end_of_list(tokens) {
         loop {
             v.push(parse_list_item(tokens)?);
-            if tokens.peek() == list_end_token {
+            if is_end_of_list(tokens) {
                 break;
             }
             consume_token(tokens, Tok::Comma)?;
-            if allow_trailing_comma && tokens.peek() == list_end_token {
+            if is_end_of_list(tokens) && allow_trailing_comma {
                 break;
             }
         }
     }
     Ok(v)
 }
 
+fn parse_comma_list_simple<'input, F, R>(
```

Comment:
I don't think it's worth splitting out a separate version of the function for this.